newIDE/app/src/EventsFunctionsExtensionEditor/index.js

```diff
@@ -498,6 +499,8 @@ export default class EventsFunctionsExtensionEditor extends React.Component<
       this._editorNavigator.openEditor('behaviors-list');
   };
 
+  static contextType = UnsavedChangesContext;
```

Comment:
Isn't this approach deprecated? Not sure it plays well with Flow/Typescript too.
We should use a Context.Consumer or a hook to get the context.

Reply:
I checked the docs and even if it is not stated that it is deprecated, a Consumer would be a better idea. Refactoring it.